### PR TITLE
Fix accidental busyloop in walkeeper's background thread

### PR DESCRIPTION
**Pull request by funbringer**

### CODE REVIEW

**Comment by funbringer** [comment]
Hope this will fix #317.

File: walkeeper/src/replication.rs [hunk 1]
```diff
@@ -51,23 +51,23 @@ impl ReplicationConn {
...
-    ///
     /// This is spawned into the background by `handle_start_replication`.
-    ///
     fn background_thread(mut stream_in: impl Read, timeline: Arc<Timeline>) -> Result<()> {
         // Wait for replica's feedback.
-        // We only handle `CopyData` messages. Anything else is ignored.
-        loop {
-            match FeMessage::read(&mut stream_in)? {
-                Some(FeMessage::CopyData(m)) => {
+        while let Some(msg) = FeMessage::read(&mut stream_in)? {
```
Comment (hlinnaka) [inline walkeeper/src/replication.rs:57]: Hmm, so the problem was that if `FeMessage::read` returned `Ok(None)`, to indicate EOF, this got stuck in the loop calling `FeMessage::read` over and over again. Is that right?

Would be good to mention in the comment above `FeMessage::read` that it returns None for EOF. And explain the problem that this fixes in a few more sentences in the commit message.

Would it be feasible to write a test or this?

Comment (funbringer) [reply walkeeper/src/replication.rs:70]: > Hmm, so the problem was that if FeMessage::read returned Ok(None), to indicate EOF, this got stuck in the loop calling FeMessage::read over and over again. Is that right?

Yes, that's it.

> Would it be feasible to write a test or this?

Okay, so I tried my best, but it feels a bit unwieldy.

> Would be good to mention in the comment above FeMessage::read that it returns None for EOF.

Sure thing. This is what the rendered docs look like (`cargo doc --open`):

image

**Review by arssher — APPROVED** [review]
Fix is good and required, but it is not clear to me why in #317 logs are not full of "INFO unexpected message".

**Comment by kelvich** [comment]
Thanks, good catch. I like this patch in it's current form.

**Pull request merged**